Library/Formula/dcadec.rb

```diff
@@ -0,0 +1,20 @@
+class Dcadec < Formula
+  desc "DTS Coherent Acoustics decoder with support for HD extensions"
+  homepage "https://github.com/foo86/dcadec"
+  url "https://github.com/foo86/dcadec/archive/v0.1.0.tar.gz"
+  sha256 "52a68a63f79ae50e5e5fd9c631a715e9feeed5d2b4a8442f0b9208692012a43f"
+  head "https://github.com/foo86/dcadec.git"
+
+  def install
+    system "make", "all"
+    unless File.exist?("test/samples/README")
+      system "git", "clone", "https://github.com/foo86/dcadec-samples.git", "test/samples"
+    end
+    system "make", "check"
+    system "make", "PREFIX=#{prefix}", "install"
+  end
+
+  test do
+    system "#{bin}/dcadec", "-h"
```

Comment:
I'm not sure how best to go about this. The package's build system has a built-in way to decode a number of provided samples (from that submodule URL) and check that the output was correct. I could duplicate that test logic in Ruby, but  it'd probably be easier to just checkout the repo and run `make check`. Would that be preferred over having `make check` run in the install step? (It'd probably be convenient to have some way of specifying that there's a test suite within the repo)